src/compiler/scala/tools/nsc/backend/jvm/opt/LocalOpt.scala

```diff
@@ -924,6 +924,47 @@ object LocalOptImpls {
       case _ => false
     })
 
+    /**
+      * Replace conditional jump instructions with GOTO or NOP if statically known to be true or false.
+      *
+      * {{{
+      *      ICONST_0; IFEQ l;
+      *   => ICONST_0; POP; GOTO l;
+      *
+      *      ICONST_1; IFEQ l;
+      *   => ICONST_1; POP;
+      * }}}
+      *
+      * Note that the LOAD/POP pairs will be removed later by `eliminatePushPop`, and the code between
+      * the GOTO and `l` will be removed by DCE (if it's not jumped into from somewhere else).
+      */
+    def simplifyConstantConditions(instruction: AbstractInsnNode): Boolean = {
+      def replace(jump: JumpInsnNode, success: Boolean): Boolean = {
+        if (success) method.instructions.insert(jump, new JumpInsnNode(GOTO, jump.label))
+        replaceJumpByPop(jump)
+        true
+      }
+
+      (instruction, previousExecutableInstruction(instruction, jumpTargets)) match {
```

Comment:
it seems we don't need to call `previousExecutableInstruction` if `instruction` is not a `ConditionalJump`.